typescript/src/commands/Moderation/lockdown.ts

```diff
@@ -62,37 +64,45 @@ export class UserCommand extends SkyraCommand {
 		}
 
 		// Create the timeout
-		const timeout = duration ? new PreciseTimeout(duration) : null;
-		message.guild.security.lockdowns.set(channel.id, { timeout });
+		const timeout = duration ? setAccurateTimeout(() => floatPromise(this._unlock(message, args.t, role, channel, allowed)), duration) : null;
+		message.guild.security.lockdowns.add(role, channel, { allowed, timeout });
+	}
 
-		// Perform cleanup later
-		if (timeout) {
-			await timeout.run();
-			await this._unlock(message, args.t, channel);
-		}
+	private isAllowed(role: Role, channel: TextChannel): boolean | null {
+		return channel.permissionOverwrites.get(role.id)?.allow.has(Permissions.FLAGS.SEND_MESSAGES, false) ?? null;
 	}
 
 	private async handleUnlock(message: GuildMessage, args: SkyraCommand.Args, role: Role, channel: TextChannel) {
 		const entry = this.getLock(role, channel);
 		if (entry === null) this.error(LanguageKeys.Commands.Moderation.LockdownUnlocked, { channel: channel.toString() });
-		return entry.timeout ? entry.timeout.stop() : this._unlock(message, args.t, channel);
+		if (entry.timeout) clearAccurateTimeout(entry.timeout);
+		return this._unlock(message, args.t, role, channel, entry.allowed);
 	}
 
-	private async _unlock(message: GuildMessage, t: TFunction, channel: TextChannel) {
-		channel.guild.security.lockdowns.delete(channel.id);
-		await channel.updateOverwrite(channel.guild.id, { SEND_MESSAGES: true });
-		return message.send(t(LanguageKeys.Commands.Moderation.LockdownOpen, { channel: channel.toString() }));
-	}
+	private async _unlock(message: GuildMessage, t: TFunction, role: Role, channel: TextChannel, allowed: boolean | null) {
```

Comment:
Give this a different name so it doesn't need an underscore prefix just to be unique from the unlock subcommand